lib/vquic/curl_quiche.c

```diff
@@ -751,6 +751,12 @@ static CURLcode cf_flush_egress(struct Curl_cfilter *cf,
   struct read_ctx readx;
   size_t pkt_count, gsolen;
 
+  quiche_conn_on_timeout(ctx->qconn);
```

Comment:
This calls the function unconditionally on every egress handling. Which is not what the quiche example seem to mandate (guessing through their lack of documentation).

I think we need to handle timeouts more elaborate here. But we can do this based on this PR.

Reply:
Thanks! I have made additional modifications to the code. Please check it. :)

Reply:
@LPardue @ghedo you think this makes sense?

Reply:
Few things:

* Calling quiche_conn_on_timeout() before any timeout elapsed is fine, there is no harm other than wasted cycles.
* It would be better to grab the timeout value once and only fire when expires, that query it every time and check for 0 (optimization)
* its not clear to me how this relates to the existing `timeout_ns` that also reads `quiche_conn_timeout()` for setting a Curl_expire

Reply:
+ That's right, using a variable to record the value of timeout can improve performance. However, in order to remain consistent with `curl_ngtcp2.c` and minimize code modifications, I chose not to do this. Maybe we can consider optimizing the implementations of `curl_ngtcp2` and `curl_quiche` together in the future.

+ Reading `quiche_conn_timeout` again is to promptly update the timing for the next timeout. After processing with `quiche_conn_on_timeout`, the timeout duration might change. Perhaps I can make a small optimization by placing the update of timeout after calling `quiche_conn_on_timeout`, which would require querying `quiche_conn_timeout` only when necessary.